dali/operators/image/convolution/laplacian.cc

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "dali/core/static_switch.h"
+#include "dali/kernels/imgproc/convolution/laplacian_cpu.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/operators/image/convolution/laplacian.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+
+using namespace convolution_utils;  // NOLINT
+
+DALI_SCHEMA(Laplacian)
+    .DocStr(R"code(Computes laplacian of the input.
+
+Laplacian is calculated as a sum of second order partial derivatives in each spatial
+dimension of the data. Each partial derivative is approximated with a separable convolution,
+that uses derivative window in the direction of the partial derivative and smoothing windows in
+the remaining axes.
+
+In total, for ``n`` spatial dimensions, there are ``n * n`` window sizes.
+The window size parameter can be:
+  * A single value used for all windows.
+  * ``n * n`` values, where ``window_size[i][j]`` is a size of window applied along ``j``-th axis
+    while computing ``i``-th partial derivative. Values ``window_size[i][i]`` describe
+    derivative window sizes, the remianing values refer to smoothing windows.
+  * ``n`` values, where the first value describes the derivative window size and
+    remaining values refer to smoothing windows. Values are shifted to the right to
+    obtain parameters for all partial derivatives. For instance, for volumetric data, specifing
+    ``window_sizes = [w1, w2, w3]`` is same as setting
+    ``window_sizes = [w1, w2, w3, w3, w1, w2, w2, w3, w1]``.
+
+Window sizes must be odd. Size of a derivative window must be at least 3. Smoothing
+window can be of size 1, which implies no smoothing along corresponding axis.
+
+To normalize output, derivative kernel that uses ``n`` windows of total size equal to ``s``
+should be scaled by 2^(-s + n + 2). If ``normalize`` argument is set to True,
+normalization is done by the operator. Alternatively, you can specify ``scale`` argument
+to customize scaling factors. Scale can be either a single value or ``n`` values, one for every
+partial derivative.
```

Comment:
I had troubles understanding that a bit, maybe this would be clearer:
```suggestion
To normalize output ``normalize=True`` can be used. Each partial derivative is scaled by ``2^(-s + n + 2)``, where ``s`` is the sum of window sizes used [by convolution] to calculate this partial derivative (including the smoothing windows) and ``n`` is the number of data dimensions/axes. Alternatively, you can specify ``scale`` argument
to customize scaling factors. Scale can be either a single value or ``n`` values, one for every
partial derivative.
```

Why is the scaling not on by default?

Reply:
Because in OpenCV it is not.

Reply:
But maybe we shouldn't rely on that this much.

Reply:
Rephrased that docs part.

Reply:
Docs looks better, I'm still not sure how to proceed with the defaults.